### PR TITLE
feat: Update plugin manager to support versions

### PR DESCRIPTION
**Pull request by roneli**
- make plugin manager support plugin creation with version
- remove usage of config package in registry
- added tests

### CODE REVIEW

File: pkg/plugin/manager.go [hunk 1]
```diff
@@ -88,44 +89,77 @@ func (m *Manager) DownloadProviders(ctx context.Context, providers []*config.Req
...
 	}
 	return p, nil
 }
 
-func (m *Manager) GetPluginDetails(providerName string) (registry.ProviderDetails, error) {
-	details, ok := m.providers[providerName]
-	if !ok {
-		return registry.ProviderDetails{}, fmt.Errorf("provider %s doesn't exist", providerName)
+// GetPluginDetails returns plugin details based on provider name
+// TODO: depercate this method
```
Comment (disq) [inline pkg/plugin/manager.go:117]:
```suggestion
// TODO: deprecate this method
```

File: pkg/client/purge.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"sort"
 	"time"
 
+	"github.com/cloudquery/cloudquery/pkg/plugin/registry"
+
```
Comment (disq) [inline pkg/client/purge.go:9]: rogue space

File: pkg/client/purge.go [hunk 3]
```diff
@@ -78,14 +80,14 @@ func PurgeProviderData(ctx context.Context, storage Storage, plugin *plugin.Mana
 	return &result, diags
 }
 
-func removeProviderStaleData(ctx context.Context, storage execution.Storage, plugin *plugin.Manager, provider string, lastUpdateTime time.Time, dryRun bool) (map[string]int, int, error) {
+func removeProviderStaleData(ctx context.Context, storage execution.Storage, plugin *plugin.Manager, provider registry.Provider, lastUpdateTime time.Time, dryRun bool) (map[string]int, int, error) {
```
Comment (disq) [inline pkg/client/purge.go:83]: `plugin *plugin.Manager` maybe make it `manager *plugin.Manager`?
Could also make an interface `ProviderSchemaGetter` etc? Not sure.

File: pkg/client/schema.go [hunk 4]
```diff
@@ -3,34 +3,36 @@ package client
...
-	providerPlugin, err := plugin.CreatePlugin(request.Provider, "", nil)
+func GetProviderSchema(ctx context.Context, manager *plugin.Manager, request *GetProviderSchemaOptions) (*ProviderSchema, error) {
+	providerPlugin, err := manager.CreatePlugin(&plugin.CreationOptions{Provider: request.Provider})
 	if err != nil {
-		log.Error().Str("provider", request.Provider).Err(err).Msg("failed to create provider plugin")
+		log.Error().Stringer("provider", request.Provider).Err(err).Msg("failed to create provider plugin")
 		return nil, err // TODO: should be a diag
 	}
-	defer plugin.ClosePlugin(providerPlugin)
+	defer manager.ClosePlugin(providerPlugin)
```
Comment (disq) [inline pkg/client/schema.go:29]: Can we have the plugin implement `io.Closer`? so this would be `defer providerPlugin.Close()`

**Review by disq — COMMENTED** [review]
nits

File: pkg/client/client.go [hunk 5]
```diff
@@ -1165,3 +1201,13 @@ func reportNumProviders(ctx context.Context, provs []*config.Provider) {
 		})
 	})
 }
+
+func ParseProviderSource(requestedProvider *config.RequiredProvider) (string, string, error) {
```
[the remark this answers is not in the document]
Comment (roneli) [reply pkg/client/client.go:1205]: This is temporary, and hopefully will be removed by the end.

File: pkg/client/client.go [hunk 6]
```diff
@@ -972,6 +993,21 @@ func (c *Client) getProviderConfig(providerName string) (*config.RequiredProvide
 	return providerConfig, nil
 }
 
+func (c *Client) CreatePlugin(providerName, alias string, env []string) (plugin.Plugin, error) {
```
[the remark this answers is not in the document]
Comment (roneli) [reply pkg/client/client.go:996]: Since this is the old version, new command call manager directly, eventually this function will be removed.

**Pull request merged**